### ISSUE TITLE
SHOC v1 input grid size does not have correct units

### ISSUE THREAD

**Issue by bogensch** [E3SM-Project/scream#1588]
SHOC requires an estimation of the local grid size (dx & dy) to properly set the maximum allowable value for the dissipation length scale.  In v1 it is being fed in with improper units and needs to be converted to meters.

The offending line of code, line 159 assumes that "area" is in meters when it is in fact in radians:
https://github.com/E3SM-Project/scream/blob/0b56109b076bf0ca76aed88824edfcb31480515c/components/scream/src/physics/shoc/atmosphere_macrophysics.hpp#L155-L159

Lines of code (or a routine) is needed to convert area from radians to meters.  One can use the subroutine "grid_size", that v0 uses (in shoc_intr.F90), as a reference for this:
https://github.com/E3SM-Project/scream/blob/0b56109b076bf0ca76aed88824edfcb31480515c/components/eam/src/physics/cam/shoc_intr.F90#L1171-L1204

**Comment by AaronDonahue** [comment]
@bogensch , so based on this, once we get the proper `grid_dx` and `grid_dy` we will want to replace `cell_length(i)=sqrt(area(i));` with `cell_length(i)=grid_dx(i)`, right?  Since we assume dx=dy?

**Comment by bogensch** [comment]
correct

**Comment by AaronDonahue** [comment]
@bogensch and @PeterCaldwell , can I get your feedback on something for this problem.  It looks like we do store an estimate of area in meters defined as follows:
```
  // Estimate cell area for a uniform grid by taking the surface area
  // of the earth divided by the number of columns
  const Real rearth    = C::r_earth;
  const Real pi        = C::Pi;
  const Real cell_area = 4*pi*rearth*rearth/num_my_cols;
  ```
We are using this in `shoc_main` but not in the `hpp` file that @bogensch notes above.

My first question is, should we be using the grid area estimate that Peter cites above from the F90 code, instead of this approximation?
Follow-up, given our deadline, would it be acceptable for now to just use this approximation for now, which would be a one line change, and then adopt the more advanced approach in the future? 

Just want to make sure I use my time wisely.

**Comment by AaronDonahue** [comment]
Note, before I found this I was able to port the F90 routine to cpp and test it.  But it would take a little more work to make sure the `hpp` has access to the latitude needed for the calculation.

**Comment by AaronDonahue** [comment]
I may also be able to redefine `cell_area` to use the more advanced formula...

**Comment by AaronDonahue** [comment]
After a little digging I see now that the `area` used in the hpp file is in fact this `cell_area` calculation.  So it should be in the units of meters like needed.  Unless something is getting mixed up in translation.  I'll have to keep digging and maybe printing values to see whats going on.

**Comment by PeterCaldwell** [comment]
I don't like the "surface area of earth divided by number of columns" approach. Columns aren't equal sized - which could cause checkerboarding or weird stuff - and this approximation would totally not work for RRM cases. I'd rather use the more advanced formula everywhere.

**Comment by AaronDonahue** [comment]
Yeah, I certainly agree with that.  I wanted to ask for short-term vs. long-term.  But I'm formulating a strategy for using the advanced formula.  The trick is getting the area in radians part right.  

In the meantime, I printed to screen what the area and length is for the use in the `hpp` and it is the definition above with the surface or earth divided by columns formula.  So it's certainly not in radians.

**Comment by PeterCaldwell** [comment]
huh. Peter B told me that empirically, the length scale was way too small. But I agree that the simple formula should definitely be in meters (unless R_earth is in km or something weird).

**Comment by AaronDonahue** [comment]
I'll note, for the print statements above I was looking at a standalone case.  It is possible in CIME cases something else is done.

**Comment by PeterCaldwell** [comment]
I don't see any print statements above but agree that the CIME case is the one to check. Especially since standalone SHOC tests never triggered this problem.

**Comment by AaronDonahue** [comment]
sorry, I meant the ones I referred to in my comment.

**Comment by AaronDonahue** [comment]
yes, I see for CIME cases we gather the area from somewhere else, which must be the default radians.

**Comment by AaronDonahue** [comment]
Fixed with #1598 , closing this now.
